curvefs/src/client/fuse_client.cpp

```diff
@@ -849,5 +1256,40 @@ void FuseClient::FlushAll() {
     FlushInodeAll();
 }
 
+CURVEFS_ERROR FuseClient::UpdateParentInodeXattr(
+    const std::list<uint64_t> &parentIds,
+    const XAttr &xattr, bool direction) {
+    for (const auto &parentId : parentIds) {
+        VLOG(1) << "UpdateParentInodeXattr inodeId = " << parentId
+                << ", \nxattr = " << xattr.DebugString();
+        std::shared_ptr<InodeWrapper> pInodeWrapper;
+        CURVEFS_ERROR ret = inodeManager_->GetInode(parentId, pInodeWrapper);
+        if (ret != CURVEFS_ERROR::OK) {
+            LOG(ERROR) << "UpdateParentInodeXattr get parent inode fail, ret = "
+                    << ret << ", inodeid = " << parentId;
+            return ret;
+        }
+        // if dirty, flush attr first
+        if (pInodeWrapper->Dirty()) {
+            ret = pInodeWrapper->SyncAttr();
+            if (ret != CURVEFS_ERROR::OK) {
+                LOG(ERROR) << "sync parent inode attr failed when"
+                           << " UpdateParentInodeXattr, inodeId = " << parentId;
+                return ret;
+            }
+        }
+        ::curve::common::UniqueLock lgGuard = pInodeWrapper->GetUniqueLock();
+        auto inode = pInodeWrapper->GetMutableInodeUnlocked();
+        for (const auto &it : xattr.xattrinfos()) {
+            auto iter = inode->mutable_xattr()->find(it.first);
+            if (iter != inode->mutable_xattr()->end()) {
+                AddUllStringToFirst(&(iter->second), it.second, direction);
+            }
```

Comment:
if iter == inode->mutable_xattr()->end();
do not need add this attr ?

Reply:
> if iter == inode->mutable_xattr()->end(); do not need add this attr ?

The summary xattr of dir will auto add when create, the xattr is not a valid summary xattr if not found.

Reply:
judge the return code

Reply:
> judge the return code

fixed